src/main/java/org/openstreetmap/atlas/geography/Polygon.java

```diff
@@ -254,18 +254,23 @@ public boolean fullyGeometricallyEncloses(final Rectangle rectangle)
             // The item is not within the bounds of this Polygon
             return false;
         }
-        // The item is within the bounds of this Polygon
-        // if this value overflows, use JTS to correctly calculate covers
+        // The item is within the bounds of this Polygon. if this value overflows
+        // use JTS to correctly calculate covers
         if (awtOverflows())
         {
             final org.locationtech.jts.geom.Polygon polygon = JTS_POLYGON_CONVERTER.convert(this);
             return polygon.covers(JTS_POLYGON_CONVERTER.convert(rectangle));
         }
-        // for most cases use the faster awt covers
-        else
+
+        // If AWT contains is false, then we want to recheck with JTS as it's less performant
+        // but more accurate-- AWT contains returns false if the calculation is too expensive
+        if (!awtArea().contains(rectangle.asAwtRectangle()))
```

Comment:
It seems this does the same thing, i.e. check JTS even if an area is really far from the Rectangle, no?

Reply:
Aaah yes, it is already checked before!